### PR TITLE
Clean up composition root

### PR DESCRIPTION
**Pull request by bojanrajkovic**

### CODE REVIEW

File: RepoInspector/RepoInspector/IO/FilesystemDataProvider.cs [hunk 1]
```diff
@@ -12,39 +13,32 @@ public class FilesystemDataProvider :
         IPullRequestCacheManager,
         IAnalysisManager
     {
+        private const string PullRequestDetails = "pull-requests.json";
+        private const string AnalysisSuffix = "-analysis.json";
+
```
Comment (rianjs) [inline RepoInspector/RepoInspector/IO/FilesystemDataProvider.cs:18]: You changed the code style here.

File: RepoInspector/RepoInspector/Records/PullRequestConstants.cs [hunk 2]
```diff
@@ -1,11 +1,17 @@
 ﻿using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
```
Comment (rianjs) [inline RepoInspector/RepoInspector/Records/PullRequestConstants.cs:2]: nice

File: RepoInspector/RepoInspector.Runner/Program.cs [hunk 3]
```diff
@@ -138,139 +116,72 @@ static async Task Main(string[] args)
...
-            _logger.LogInformation($"Daemon stopped in {stopwatch.ElapsedMilliseconds:N0}ms");
-        }
-        
-        private static IScorerFactory GetKnownScorers(IApprovalAnalyzer approvalAnalyzer, IWordCounter wc)
+        private static IScorerFactory GetKnownScorers(IServiceProvider sp)
         {
             var scorers = GetDerivedTypes<Scorer>(Assembly.GetAssembly(typeof(Scorer)));
             var scorerInstances = scorers
-                .Select(s => CreateInstance(s, approvalAnalyzer, wc))
+                .Select(s => ActivatorUtilities.GetServiceOrCreateInstance(sp, s))
```
Comment (rianjs) [inline RepoInspector/RepoInspector.Runner/Program.cs:172]: Am I crazy? Where is this implementation?

File: RepoInspector/RepoInspector/IO/FilesystemDataProvider.cs (hunk 1 above)
Comment (bojanrajkovic) [reply RepoInspector/RepoInspector/IO/FilesystemDataProvider.cs:18]: Hmm, yep, you're right.

File: RepoInspector/RepoInspector.Runner/Program.cs (hunk 3 above)
Comment (bojanrajkovic) [reply RepoInspector/RepoInspector.Runner/Program.cs:172]: Yeah, that's why this is WIP. `ActivatorUtilities` is provided by the framework, and I *think* there's a way to make it do the magic, but the whole instantiation of scorers is a little unclear to me -- it seems like given the current setup, they'll always get GitHub-flavored parsers?

Comment (rianjs) [reply RepoInspector/RepoInspector.Runner/Program.cs:172]: * Scorers are repo-type agnostic. These are all stateless singletons.
* PR readers are repo-type-specific, and are 1:1 with a `WatchedRepository`.

Each PR reader is instantiated by the `PullRequestReaderFactory`, looking at the type of `RepositoryKind` in each `WatchedRepository`, passed in from config.

**Pull request closed without merging**